src/widgets/search-box/search-box.tsx

```diff
@@ -0,0 +1,263 @@
+  }
+>;
+
+export type SearchBoxWidgetParams = {
+  /**
+   * CSS Selector or HTMLElement to insert the widget
+   */
+  container: string | HTMLElement;
+  /**
+   * The placeholder of the input
+   */
+  placeholder?: string;
+  /**
+   * Whether the input should be autofocused
+   */
+  autofocus?: boolean;
+  /**
+   * If set, trigger the search
+   * once `<Enter>` is pressed only.
+   */
+  searchAsYouType?: boolean;
+  /**
+   * Whether to show the reset button
+   */
+  showReset?: boolean;
+  /**
+   * Whether to show the submit button
+   */
+  showSubmit?: boolean;
+  /**
+   * Whether to show the loading indicator (replaces the submit if
+   * the search is stalled)
+   */
+  showLoadingIndicator?: boolean;
+  /**
+   * CSS classes to add
+   */
+  cssClasses?: Partial<SearchBoxCSSClasses>;
+
+  /**
+   * Templates used for customizing the rendering of the searchbox
+   */
+  templates?: Partial<SearchBoxTemplates>;
+  /**
+   * A function that is called every time a new search is done. You
+   * will get the query as the first parameter and a search (query) function to call as the second parameter.
+   * This `queryHook` can be used to debounce the number of searches done from the search box.
+   */
+  queryHook?: (query: string, hook: (value: string) => void) => void;
+};
+
+const renderer = ({
+  containerNode,
+  cssClasses,
+  placeholder,
+  templates,
+  autofocus,
+  searchAsYouType,
+  showReset,
+  showSubmit,
+  showLoadingIndicator,
+}: {
+  containerNode: HTMLElement;
+  cssClasses: SearchBoxRendererCSSClasses;
+  placeholder: string;
+  templates: SearchBoxTemplates;
+  autofocus: boolean;
+  searchAsYouType: boolean;
+  showReset: boolean;
+  showSubmit: boolean;
+  showLoadingIndicator: boolean;
+}) => ({
+  refine,
+  query,
+  isSearchStalled,
+}: SearchBoxRendererOptions & RendererOptions<SearchBoxConnectorParams>) => {
+  render(
+    <SearchBox
+      query={query}
+      placeholder={placeholder}
+      autofocus={autofocus}
+      refine={refine}
+      searchAsYouType={searchAsYouType}
+      templates={templates}
+      showSubmit={showSubmit}
+      showReset={showReset}
+      showLoadingIndicator={showLoadingIndicator}
+      isSearchStalled={isSearchStalled}
+      cssClasses={cssClasses}
+    />,
+    containerNode
+  );
+};
+
+/**
+ * The searchbox widget is used to let the user set a text based query.
+ *
+ * This is usually the  main entry point to start the search in an instantsearch context. For that
+ * reason is usually placed on top, and not hidden so that the user can start searching right
+ * away.
+ *
+ * @type {WidgetFactory}
+ * @devNovel SearchBox
+ * @category basic
+ * @param {SearchBoxWidgetParams} widgetParams Options used to configure a SearchBox widget.
+ * @return {Widget} Creates a new instance of the SearchBox widget.
+ * @example
+ * search.addWidgets([
+ *   instantsearch.widgets.searchBox({
+ *     container: '#q',
+ *     placeholder: 'Search for products',
+ *   })
+ * ]);
```

Comment:
```suggestion
```

I think the other widgets no longer have this block, but it's tracked in a separate ticket, so fine either way